applications/callflow/src/cf_route_win.erl

```diff
@@ -237,14 +237,18 @@ store_owner_id(Call) ->
 -spec set_language(whapps_call:call()) -> whapps_call:call().
 set_language(Call) ->
     Default = wh_media_util:prompt_language(whapps_call:account_id(Call)),
-    case cf_endpoint:get(Call) of
-        {'ok', Endpoint} ->
-            Language = kz_device:language(Endpoint, Default),
-            lager:debug("setting language '~s' for this call", [Language]),
-            whapps_call:set_language(wh_util:to_lower_binary(Language), Call);
-        {'error', _E} ->
-            lager:debug("no source endpoint for this call, setting language to default ~s", [Default]),
-            whapps_call:set_language(Default, Call)
+    case whapps_call:language(Call) of
+        Default ->
+            case cf_endpoint:get(Call) of
+                {'ok', Endpoint} ->
+                    Language = kz_device:language(Endpoint, Default),
+                    lager:debug("setting language '~s' for this call", [Language]),
+                    whapps_call:set_language(wh_util:to_lower_binary(Language), Call);
+                {'error', _E} ->
+                    lager:debug("no source endpoint for this call, setting language to default ~s", [Default]),
+                    whapps_call:set_language(Default, Call)
```

Comment:
@danielfinke 
Ah ok I understand now, thanks :)

SO I would prefer Luis's comment on the other PR:

"yes, i understand the path now. thank you.
maybe we should change how resume works because it still execs the other actions in bootstrap.
let me run some tests."

@lazedo you left it at "let me run some tests", let us know what you find :)

My other thought would be to ensure the language is not set unless explicitly provided and all functions that access this information fallback to the default if not present (perhaps even in whapps_call).  This way 'undefined' means 'not set use default', and any other value means 'this is what the user wants'.  If we did this then I would feel much better about ignoring the language if set in route_win.